tensorflow/core/lib/strings/str_util.cc

```diff
@@ -416,6 +417,99 @@ bool ConsumeNonWhitespace(StringPiece* s, StringPiece* val) {
   }
 }
 
+Status SplitUTF8(StringPiece text, const string& delim, const bool skip_empty,
+                 std::vector<string>* result) {
+  // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
+  //   1     0xxxxxxx
+  //   2     110xxxxx  10xxxxxx
+  //   3     1110xxxx  10xxxxxx  10xxxxxx
+  //   4     11110xxx  10xxxxxx  10xxxxxx  10xxxxxx
+  result->clear();
+  size_t char_start = 0, char_len = 0;
+  bool advance = true;
+  for (size_t i = 0; i < text.size(); ++i) {
+    if (char_start == i) {
+      if (!UTF8CharNumBytes(text, &char_len)) {
+        result->clear();
+        return errors::InvalidArgument("Invalid UTF8 encoding at byte ", i);
+      }
+    }
+    // Validate all intermediate bytes are valid UTF-8
+    if (TF_PREDICT_FALSE((char_len > 1) && (char_start != i) &&
+                         (text[i] & 0xC0) != 0x80)) {
+      return errors::InvalidArgument("Invalid UTF8 encoding at byte ", i);
+    }
+    // Reached the end of a character
+    if (char_start + char_len == i + 1) {
+      const StringPiece entry = text.substr(char_start, char_len).ToString();
+      if (delim == "") {
+        result->emplace_back(entry.ToString());
+      } else if (delim == entry) {
+        advance = true;
+        if (!skip_empty) {
+          if (TF_PREDICT_FALSE(char_start == 0)) {
```

Comment:
IIUC, this is handling the case when `skip_empty` is true and the delimiter is the first or last character. But what about cases where there are multiple consecutive delimiters? See comment about testing (and perhaps duplicating/reusing `testStringSplitWithNoSkipEmpty` with this new op)

Reply:
Thanks @asimshankar. My previous interpretation was not right. I have updated the PR so that now it will match python behavior in case `skip_empty=False`, i.e.,
```
>>> "##a##b##c##".split("#")
['', '', 'a', '', 'b', '', 'c', '', '']
>>> 
```